plugin/core/sessions.py

```diff
@@ -791,8 +792,8 @@ def _template_variables(self) -> Dict[str, str]:
 
     def run_command(self, command: Mapping[str, Any]) -> Promise:
         if self._plugin:
-            promise = self._plugin.on_pre_server_command(command)
-            if promise:
+            promise, callback = Promise.packaged_task()
```

Comment:
I find the usage of `Promise.packaged_task` a bit confusing. 

If you want run a callback when the `on_pre_server_command` is done, why not write `self._plugin.on_pre_server_command(command).then(callback)`?

If we don't have the `Promise.packaged_task` I would find the code easier to understand.

Reply:
> If you want run a callback when the `on_pre_server_command` is done, why not write `self._plugin.on_pre_server_command(command).then(callback)`?

`.on_pre_server_command(command)` doesn't return a Promise so you can call `.then` on it.

Reply:
Oh, I tough it was a promise based on the old code I saw:
```diff
-            promise = self._plugin.on_pre_server_command(command)
```